### PR TITLE
Move some gradient ascent optimizer flags into the gin config

### PR DESCRIPTION
**Pull request by boomanaiden154**
This patch moves command line flags in es_trainer_lib to set the gradient
ascent optimizer and gradient ascent optimizer flags to the gin config along
with all the other model/optimizer hyperparameters. These don't make a lot of
sense to have as command line parameters given their nature and everything
else similar being defined in gin configs.

### CODE REVIEW

File: compiler_opt/es/es_trainer_lib.py [hunk 1]
```diff
@@ -60,11 +51,21 @@
                                      "List of paths to training corpora")
 
 
+@gin.constants_from_enum(module="es_trainer_lib")
+class GradientAscentOptimizerType(enum.Enum):
+  MOMENTUM = 1
```
Comment (mtrofin) [inline compiler_opt/es/es_trainer_lib.py:56]: Add a `INVALID = 0` maybe

also I think you can use `enum.auto()`

**Review by mtrofin — APPROVED** [review]
lgtm, a comment.
